event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocator.java

```diff
@@ -94,22 +95,22 @@ public SlabAllocation allocate(String channelName, int desiredCount) {
             // SlabPersister I/O operation around creating slabs.
 
             // Is there is an existing open slab we can allocate from?
-            SlabAllocation allocation = channelState.allocate(desiredCount);
+            SlabAllocation allocation = channelState.allocate(desiredCount, eventSizes);
             if (allocation != null) {
                 return allocation;
             }
 
             // No existing slab.  Create a new slab just for the caller and not shared with anyone else.
             SlabRef slab = createSlab(channelName);
-            return new DefaultSlabAllocation(slab, 0, Constants.MAX_SLAB_SIZE);
+            return new DefaultSlabAllocation(slab, 0, Constants.MAX_SLAB_SIZE, Constants.MAX_SLAB_SIZE_IN_BYTES);
```

Comment:
This is not safe anymore unfortunately. We can't allocate them a full slab because we don't know the number of events that will fit.